compiler/vm/vmconv.nim

```diff
@@ -1,5 +1,13 @@
 import ast/ast
 
+import vm/[
+  vmdef,
+  vmobjects
+]
+
+# XXX: correctly and fully supporting every type with `fromLit`/`toLit` seems
```

Comment:
I think this is what would potentially become easier with the field generalization we chatted about.

Reply:
I don't see how? The complexity/work is due to having to write a whole lot of `when T is xxx`/`elif` blocks in order to support all types representable by the VM (e.g. `array`, `seq`, `int`, `int32`, `int8`, etc.). 

Generalized fields wouldn't help there, no?

Reply:
In that I imagine the number of types that needs to be supported by the VM would decrease. Potentially we could collapse array and seq. Set might disappear too.

Although I didn't get a chance to check the latest list of types and it might already be shorter. In which case you're absolutely correct.